aiida/calculations/transfer.py

```diff
@@ -0,0 +1,211 @@
+            'Please include at least one of `local_files`, `remote_files`, or `symlink_files`.\n'
+            'These should be lists containing tupples following the pattern:\n'
+            '[ ... (source_uuid, source_relpath, target_relpath) ... ] \n'
+        )
+        return errmsg.format(instructions.uuid)
+
+    source_nodes_provided = set(source_nodes)
+    source_nodes_demanded = set()
+    if local_files:
+        source_nodes_demanded.update((file_data[0] for file_data in local_files))
+    if remote_files:
+        source_nodes_demanded.update((file_data[0] for file_data in remote_files))
+    if symlink_files:
+        source_nodes_demanded.update((file_data[0] for file_data in symlink_files))
+
+    if source_nodes_provided > source_nodes_demanded:
+        errmsg = 'some nodes provided as inputs are not being used:\n'
+        for extra_node in source_nodes_provided.difference(source_nodes_demanded):
+            errmsg = errmsg + ' > {}\n'.format(extra_node)
+        return errmsg
+
+    if source_nodes_demanded > source_nodes_provided:
+        errmsg = 'some nodes requested in the instructions are not provided as inputs:\n'
+        for extra_node in source_nodes_demanded.difference(source_nodes_provided):
+            errmsg = errmsg + ' > {}\n'.format(extra_node)
+        return errmsg
+
+
+class TransferCalculation(CalcJob):
+    """Utility to copy files from different FolderData and RemoteData nodes into a single place.
+
+    The final destination for these files can be either the local repository (by creating a
+    new FolderData node to store them) or in the remote computer (by leaving the files in a
+    new remote folder saved in a RemoteData node).
+
+    Only files from the local computer and a single remote computer can be moved with a single
+    instance of this CalcJob (so you can't directly move files from remote to remote and all
+    RemoteData nodes must be from the same machine).
+
+    The user needs to provide three inputs:
+    * `instructions`: a dict node specifying which files to copy from which nodes.
+    * `source_nodes`: a dict of nodes, each with a unique identifier label as its key.
+    * `metadata.computer`: the computer that contains the remote files and will contain the final
+    RemoteData node.
+
+    The `instructions` dict must have the `retrieve_files` flag. The CalcJob will create a new
+    folder in the remote machine (RemoteData) and put all the files there and will either:
+        (1) leave them there (`retrieve_files = False`) or ...
+        (2) create a local FolderData and copy there all the files (`retrieve_files = True`)
+
+    The `instructions` dict must also contain at least one list with specifications of which files
+    to copy and from where. All these lists take tupples of 3 that have the following format:
+
+        [ ..., ( source_node_label, path_to_file_in_source, path_to_file_in_target), ... ]
```

Comment:
Should probably make very clear that `source_node_label` corresponds to the key used in the `source_nodes` namespace